blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/credential/BlueOceanCredentialsProvider.java

```diff
@@ -0,0 +1,261 @@
+        FolderPropertyImpl prop = (FolderPropertyImpl)folder.getProperties().get(FolderPropertyImpl.class);
+        if (prop != null) {
+            User user = User.get(prop.getUser(), false, Collections.emptyMap());
+            if(user != null){
+                Authentication auth = user.impersonate();
+                return CredentialsMatchers.filter(
+                        CredentialsProvider.lookupCredentials(
+                                type,
+                                Jenkins.getInstance(),
+                                auth,
+                                Collections.<DomainRequirement>emptyList()
+                        ),
+                        CredentialsMatchers.allOf(CredentialsMatchers.withId(prop.getId()),
+                                CredentialsMatchers.withScope(CredentialsScope.USER))
+                );
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    @Nonnull
+    @Override
+    public <C extends IdCredentials> ListBoxModel getCredentialIds(@Nonnull Class<C> type,
+                                                                   @Nullable ItemGroup itemGroup,
+                                                                   @Nullable Authentication authentication,
+                                                                   @Nonnull List<DomainRequirement> domainRequirements,
+                                                                   @Nonnull CredentialsMatcher matcher) {
+        ListBoxModel result = new ListBoxModel();
+        if (itemGroup instanceof AbstractFolder) {
+            FolderPropertyImpl prop = (FolderPropertyImpl) ((AbstractFolder) itemGroup).getProperties().get(FolderPropertyImpl.class);
+            if (prop != null) {
+                result.add(Messages.BlueOceanCredentialsProvider_DisplayName(), prop.getId());
+            }
+
+        }
+        return result;
+    }
+
+    @Override
+    @Nonnull
+    public String getDisplayName() {
+        return Messages.BlueOceanCredentialsProvider_DisplayName();
+    }
+
+    @Override
+    public CredentialsStore getStore(@CheckForNull ModelObject object) {
+        if(isApplicable(object)) {
+            return new FolderPropertyImpl.StoreImpl((AbstractFolder) object);
+        }
+        return null;
+    }
+
+    @Override
+    public Set<CredentialsScope> getScopes(ModelObject object) {
+        if(isApplicable(object)){
+            return ImmutableSet.of(CredentialsScope.USER);
+        }
+        return Collections.emptySet();
+    }
+
+    private boolean isApplicable(ModelObject object){
+        return object instanceof AbstractFolder &&
+                ((AbstractFolder)object).getProperties().get(FolderPropertyImpl.class) != null;
+    }
+
+    public static class FolderPropertyImpl extends AbstractFolderProperty<AbstractFolder<TopLevelItem>> {
+        private final String user;
```

Comment:
@stephenc Yes I realize storing domain name is better along with domain specification. Is there Credentials API or somewhere I could use code that creates domain spec. I do it for github/https, bit tedious to take in to account different URI types.